src/main/java/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,18 @@
+package strategy;
+
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> inventory, String fruit, int quantity) {
```

Comment:
According to the checklist, you should check the result balance before saving it in the Storage and it should be positive. However, you are only checking if there is enough inventory to subtract the quantity. Ensure that after the operation, the balance remains positive, and throw a RuntimeException if the balance is negative.